imsim/diffraction_fft.py

```diff
@@ -0,0 +1,226 @@
+    with a linear decay with the distance to the axes."""
+    rot = np.array(
+        [[np.cos(-alpha), -np.sin(-alpha)], [np.sin(-alpha), np.cos(-alpha)]]
+    )
+    xy_rotated = np.einsum("ij,jkl", rot, xy)
+    # Antialiased cross:
+    return np.maximum(0.0, 1.0 - np.min(np.abs(xy_rotated), axis=0))
+
+
+def prepare_psf_field_rotation(
+    w: int,
+    h: int,
+    resolution: float,
+    alpha: float,
+    d_alpha: float,
+) -> np.ndarray:
+    """Spike PSF for finite angular spike with (field rotation).
+    w: Pixel width of the image
+    h: Pixel height of the image
+    resolution: length per pixel
+    alpha: Rotation angle of the spikes [rad]
+    d_alpha: angular spike width (field rotation angle)
+
+    Will return an image of the spike PSF of dimension (2w+1)x(2h+1).
+    The angular range of the spike will be [alpha, alpha + d_alpha] (4-fold symmetry).
+    """
+    x_range = np.arange(-w, w + 1)
+    y_range = np.arange(-h, h + 1)
+    [x, y] = xy = np.c_[
+        np.meshgrid(
+            x_range,
+            y_range,
+            indexing="ij",
+            copy=False,
+        )
+    ]
+    # Antialiased cross at alpha - d_alpha / 2:
+    psf = antialiased_cross(xy, alpha - d_alpha / 2.0)
+    # Interior:
+    th = np.arctan2(y, x)
+    dth = th - (alpha - d_alpha)  # angular distance to spike boundary
+    dth %= np.pi / 2  # 4-fold symmetry
+    psf[dth <= d_alpha] = 1.0
+    # Radial profile:
+    r = np.hypot(x, y) * resolution
+    psf *= field_rotation_profile(r, d_alpha, resolution)
+    # Normalize, so pixel values add up to 1.
+    # Here we actually are neglecting the non-visible tails of the cross.
+    psf /= np.sum(psf)
+    return psf
+
+
+def apply_diffraction_psf(
+    image: np.ndarray,
+    rottelpos: float,
+    exptime: np.ndarray,
+    latitude: float,
+    azimuth: float,
+    altitude: float,
+    full_well: float,
+    pixel_length: float,
+    psf_size: int,
+):
+    """2d convolve, leaving image dimension invariant.
+    image: Brightness pixel data
+    rottelpos: Telescope rotation angle [rad]
+    exptime: Exposure time (used to determine the spike width)
+    latitude: Geographic latitude of observators
+    azimuth: Azimuth of the telescope
+    altitude: Altitude of the telescope
+    full_well: Threshold value for saturated pixels
+    pixel_length: Scaling factor: The length of 1px in `spike_profile`
+    psf_size: Size of the PSF (width and height) which will be layed over saturated pixels.
+    """
+
+    psf_w = psf_h = psf_size  # neglect the outer region (~1/r^2)
+    sin_cos = np.empty(2)
+    e_z_0, e_z = prepare_e_z(latitude)
+    e_focal = e_equatorial(latitude=latitude, azimuth=azimuth, altitude=altitude)
+    field_rotation_sin_cos(e_z_0, e_z, e_focal, exptime, sin_cos)
+    d_alpha = np.arctan2(sin_cos[1], sin_cos[0])
+    rottelpos = np.pi/4. - rottelpos
+    psf = prepare_psf_field_rotation(
```

Comment:
`psf` isn't a great name fro this.  This is just the diffraction spikes, not the whole psf.  Probably `spike_image` would be better.  Or `spike_per_pixel` to point out that it's just the diffraction spikes corresponding to a single pixel in the undiffracted psf.